src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -1864,9 +1864,14 @@ void primitive_inst::prepare_primitive() {
         // Only try update weight and realloc when impl is updated.
         const bool can_use_async_compilation = use_async_compilation();
         const bool shape_changed = get_flag(ExecutionFlags::SHAPE_CHANGED);
-        if (shape_changed || !_impl || (!shape_changed && _impl->is_dynamic() && can_use_async_compilation)) {
+        // for runtime skippable nodes, if previous iter is skipped while this iter not, its output memory needs to be revalidate
+        // as memory opt/release may be applied for these nodes to reduce memory footprint in previous iters
+        const bool need_realloc_for_runtime_skippable =
+            prev_execution_skipped && !can_be_optimized() &&
+            _network.get_engine().is_the_same_buffer(output_memory(), input_memory());
+        if (shape_changed || !_impl || (!shape_changed && _impl->is_dynamic() && can_use_async_compilation) || need_realloc_for_runtime_skippable) {
```

Comment:
Maybe we can simplify it a bit? Something like::
```
        if (shape_changed || !_impl || (!shape_changed && _impl->is_dynamic() && can_use_async_compilation)) {
            ...
        } else if (need_realloc_for_runtime_skippable) {
            realloc_if_needed(prev_execution_skipped);
        }
```
It seems we don't really need to call update_impl()/update_weights() for need_realloc_for_runtime_skippable?

Reply:
ah.. yes :)